lib/web_ui/lib/src/engine/clipboard.dart

```diff
@@ -65,6 +65,30 @@ class ClipboardMessageHandler {
     });
   }
 
+  /// Handles the platform message which asks if the clipboard contains
+  /// pasteable strings.
+  void hasStringsMethodCall(ui.PlatformMessageResponseCallback? callback) {
+    const MethodCodec codec = JSONMethodCodec();
+    _pasteFromClipboardStrategy.getData().then((String data) {
+      final Map<String, dynamic> map = <String, dynamic>{'value': data.isNotEmpty};
+      callback!(codec.encodeSuccessEnvelope(map));
+    }).catchError((dynamic error) {
+      if (error is UnimplementedError) {
+        // Clipboard.hasStrings not supported.
+        // Passing [null] to [callback] indicates that the platform message isn't
+        // implemented. Look at [MethodChannel.invokeMethod] to see how [null] is
+        // handled.
+        Future<void>.delayed(Duration.zero).then((_) {
+          if (callback != null) {
+            callback(null);
+          }
+        });
+        return;
+      }
+      _reportGetDataFailure(callback, codec, error);
```

Comment:
The app is only checking whether the clipboard contains strings, it's not doing a paste. This error is going to be confusing in that case. It basically says that the "paste" operation has failed.

I suggest that we fail silently here and just return `false` to the `hasStrings` call. This covers the case when the user denies clipboard permission, in which case we should tell the app that there are no strings in the clipboard (basically treat the clipboard as if it were empty).

Reply:
Good call!  Fixed and added a test.